src/goto-programs/goto_check.cpp

```diff
@@ -342,16 +343,39 @@ void goto_checkt::bounds_check(
 
   assert(is_array_type(t) || is_string_type(t) || is_vector_type(t));
 
-  // We can't check the upper bound of an infinite sized array
```

Comment:
I suggest keeping the early return for an infinite-sized array and adding another one for an incomplete array.